client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,202 @@
+  singleFile: PropTypes.string,
+  formRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({ current: PropTypes.instanceOf(Object) })
+  ]),
+  isDisabled: PropTypes.bool.isRequired,
+  validator: PropTypes.func.isRequired
+};
+DataFilesCompressForm.defaultProps = {
+  formRef: null,
+  singleFile: null
+};
+
+const DataFilesCompressModal = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const [valid, setValid] = useState(false);
+  const dispatch = useDispatch();
+  const status = useSelector(
+    state => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const params = useSelector(
+    state => state.files.params.FilesListing,
+    shallowEqual
+  );
+
+  const isOpen = useSelector(state => state.files.modals.compress);
+  const selectedFiles = useSelector(
+    ({ files: { selected, listing } }) =>
+      selected.FilesListing.map(i => ({
+        ...listing.FilesListing[i]
+      })),
+    shallowEqual
+  );
+  const selected = useMemo(() => selectedFiles, [isOpen]);
+  const formRef = React.useRef();
+  const toggle = () =>
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'compress', props: {} }
+    });
+
+  const onOpened = () => {
+    dispatch({
+      type: 'FETCH_FILES_MODAL',
+      payload: { ...params, section: 'modal' }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
+    if (status) {
+      dispatch({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: {}, operation: 'compress' }
+      });
+      history.push(location.pathname);
+    }
+  };
+
+  const compressCallback = () => {
+    if (!valid) return;
+    const { filenameDisplay, filetype } = formRef.current.values;
+    const filename = `${filenameDisplay.replace(' ', '')}${filetype}`;
+    dispatch({
+      type: 'DATA_FILES_COMPRESS',
+      payload: { filename, files: selected }
+    });
+  };
+
+  let buttonIcon;
+  if (status === 'RUNNING') {
+    buttonIcon = <LoadingSpinner placement="inline" />;
+  } else if (status === 'ERROR') {
+    buttonIcon = <Icon name="alert" />;
+  } else {
+    buttonIcon = null;
+  }
+  const firstFile = selectedFiles[0] ? selectedFiles[0].name : '';
+  return (
+    <Modal
+      isOpen={isOpen}
+      onOpened={onOpened}
+      onClosed={onClosed}
+      toggle={toggle}
+      className="dataFilesModal"
+    >
+      <ModalHeader toggle={toggle}>Compress Files</ModalHeader>
+      <ModalBody>
+        <DataFilesCompressForm
+          formRef={formRef}
+          singleFile={selectedFiles.length > 1 ? null : firstFile}
+          isDisabled={status === 'RUNNING' || status === 'SUCCESS'}
+          validator={setValid}
+        />
+        <p>
+          A job to compress your files will be submitted on your behalf. You can
+          check the status of this job on your Dashboard, and your compressed
+          file will appear in this directory.
```

Comment:
**Text Change** (from Design)

There has been this text change defined in [a message in private channel `cep-design-internal`](https://tacc-team.slack.com/archives/G01K9ET7JM7/p1610671076006400).

```suggestion
          A job to compress these files will be submitted. The compressed file
          will appear in this directory.
```